src/util/GuardedStream.ts

```diff
@@ -80,23 +61,29 @@ function removeDefaultErrorListener(this: Guarded, event: string): void {
     const errors = this[guardedErrors];
     if (errors.length > 0) {
       this[guardedErrors] = [];
-      setImmediate((): void => {
-        for (const error of errors) {
+      for (const error of errors) {
+        setImmediate((): void => {
           this.emit('error', error);
-        }
-      });
+        });
+      }
```

Comment:
Actually the order in the original was a deliberate choice: fire all the errors at once, with no time in between for changing error listeners etc. Otherwise, you might be re-catching an error you've emitted yourself. And this might mess with `errors.length`, etc.

Reply:
Makes sense. I was thinking of one of the error listeners crashing and thereby preventing the other ones from firing.